integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -113,7 +114,29 @@ fn test_resharding_v3() {
         let prev_epoch_id =
             client.epoch_manager.get_prev_epoch_id_from_prev_block(&tip.prev_block_hash).unwrap();
         let epoch_config = client.epoch_manager.get_epoch_config(&prev_epoch_id).unwrap();
-        epoch_config.shard_layout.shard_ids().count() == expected_num_shards
+        if epoch_config.shard_layout.shard_ids().count() != expected_num_shards {
+            return false;
+        }
+
+        // If resharding happened, also check that each shard has non-empty state.
+        for shard_id in 0..3 {
+            let shard_uid = ShardUId { version: 3, shard_id: shard_id as u32 };
+            let chunk_extra =
+                client.chain.get_chunk_extra(&tip.prev_block_hash, &shard_uid).unwrap();
+            let trie = client
+                .runtime_adapter
+                .get_trie_for_shard(
+                    shard_id,
+                    &tip.prev_block_hash,
+                    *chunk_extra.state_root(),
+                    false,
+                )
+                .unwrap();
+            let items = trie.lock_for_iter().iter().unwrap().count();
+            assert!(items > 0);
+        }
+
+        return true;
     };
 
     test_loop.run_until(
```

Comment:
Not for this PR but I would be tempted to pretty print the state (perhaps accounts only) before and after resharding.